ipaserver/install/ipa_cacert_manage.py

```diff
@@ -466,6 +479,66 @@ def list(self):
         for _ca_cert, ca_nickname, _ca_trust_flags in ca_certs:
             print(ca_nickname)
 
+    def delete(self):
+        options = self.options
+        nickname = self.args[1]
+        conn = api.Backend.ldap2
+
+        ca_certs = certstore.get_ca_certs_nss(api.Backend.ldap2,
+                                              api.env.basedn,
+                                              api.env.realm,
+                                              False)
+
+        ipa_ca_nickname = get_ca_nickname(api.env.realm)
+
+        found = False
+        for _ca_cert, ca_nickname, _ca_trust_flags in ca_certs:
+            if ca_nickname == nickname:
+                found = True
+                break
+
+        if not found:
+            raise admintool.ScriptError(
+                'Unknown CA \'{}\''.format(nickname)
+            )
+
+        with certs.NSSDatabase() as tmpdb:
+            tmpdb.create_db()
+            for ca_cert, ca_nickname, ca_trust_flags in ca_certs:
+                tmpdb.add_cert(ca_cert, ca_nickname, ca_trust_flags)
+            loaded = tmpdb.list_certs()
+            logger.debug("loaded raw certs '%s'", loaded)
+
+            tmpdb.delete_cert(nickname)
+
+            for ca_nickname, _trust_flags in loaded:
+                if ca_nickname == nickname:
```

Comment:
This check should be done after the check for ipa_ca_nickname == nickname (otherwise it's possible to remove IPA CA).

Reply:
Strange, I had at test for that at some point but lost it somehow. Anyway, added.

I suspect there may be valid reasons to do this so I set the error to be "cannot be deleted with this tool". Too dangerous.